pkg/clusteragent/admission/mutate/autoinstrumentation/language_versions.go

```diff
@@ -224,6 +230,14 @@ func (i libInfo) initContainers(v version) []initContainer {
 		command = []string{"sh", "copy-lib.sh", mounts[0].MountPath}
 	}
 
+	if i.imageResolver != nil {
+		log.Debugf("Resolving image %s/%s:%s", i.registry, i.repository, i.tag)
+		image, ok := i.imageResolver.Resolve(i.registry, i.repository, i.tag)
+		if ok {
+			i.image = image.FullImageRef
```

Comment:
This should apply the same pattern 
https://github.com/DataDog/datadog-agent/pull/40762/files#diff-75a60dbf5546cab4c0507ad10f27369dae56c856c860443129282f7dba73282cL60-R60